File: problems/0128-longest-consecutive-sequence/solution.go
// Link: https://leetcode.cn/problems/longest-consecutive-sequence/
// Level: Medium
// Date: 2025-05-10

func longestConsecutive(nums []int) int {
	numMap := make(map[int]bool)
	for _, num := range nums {
		numMap[num] = true
	}
	maxLen := 0
	for num := range numMap {
		if !numMap[num-1] {
			currNum := num
			currLen := 1
			for numMap[currNum+1] {
				currNum++
				currLen++
			}
			if currLen > maxLen {
				maxLen = currLen
			}
		}
	}
	return maxLen
}

File: problems/0128-longest-consecutive-sequence/solution.cpp
// Link: https://leetcode.cn/problems/longest-consecutive-sequence/
// Level: Medium
// Date: 2025-05-10

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> num_set(nums.begin(), nums.end());
        int max_len = 0;
        for (int num : num_set) {
            if (!num_set.count(num - 1)) {
                int current = num, current_len = 1;
                while (num_set.count(current + 1)) {
                    current++;
                    current_len++;
                }
                max_len = max(max_len, current_len);
            }
        }
        return max_len;
    }
};
